.github/workflows/release_ruleset.yaml

```diff
@@ -55,31 +74,76 @@ jobs:
 
-      - name: Build release artifacts and prepare release notes
+      - name: Release preparation
+        id: release_prep
+        working-directory: this
         run: |
           if [ ! -f "${{ inputs.release_prep_command }}" ]; then
             echo "ERROR: create a ${{ inputs.release_prep_command }} release prep script or configure a different release prep command with the release_prep_command attribute"
             exit 1
           fi
-          ${{ inputs.release_prep_command }} ${{ inputs.tag_name || github.ref_name }} > release_notes.txt
+
+          export ARTIFACTS_DIR=$(mktemp --directory)
+          export RELEASE_NOTES=$(mktemp)
+          OUTPUT=$(${{ inputs.release_prep_command }} ${{ inputs.tag_name || github.ref_name }} | jq --compact-output .)
+
+          echo "Release prep output:"
+          echo "${OUTPUT}"
+
+          # Parse the the release archives, making the paths absolute
+          # Store a comma-delimited list which is the format required by the `subject-path`
+          # input in actions/attest-build-provenance.
+          RELEASE_ARCHIVES=$(echo "${OUTPUT}" | jq --raw-output --arg artifactsDir "${ARTIFACTS_DIR}/" '.release_archives[] |= sub("^(?<path>[^/].*)";"\($artifactsDir)\(.path)") | .release_archives[]' | tr '\n' ',' | sed '$s/,$//')
+
+          echo "artifacts_dir=$ARTIFACTS_DIR" >> $GITHUB_OUTPUT
+          echo "release_notes=$RELEASE_NOTES" >> $GITHUB_OUTPUT
+          echo "release_archives=$RELEASE_ARCHIVES" >> $GITHUB_OUTPUT
+
+      # The actions/attest-build-provenance action can only produce a single attestation with multiple
+      # subjects, rather than an attestation per subject, which we want. Create a single attestation
+      # with all release archives, then copy the same attestation to multiple files below.
+      - name: Attest release archive(s) provenance
+        id: attest_release_archive
+        uses: actions/attest-build-provenance@v2
+        with:
+          subject-path: ${{ steps.release_prep.outputs.release_archives }}
+      
+      - name: Write release archive(s) attestation into intoto.jsonl
+        id: write_release_archive_attestation
+        run: |
+          ATTESTATIONS_DIR=$(mktemp --directory)
+          RELEASE_ARCHIVES="${{ steps.release_prep.outputs.release_archives }}"
+          SPACE_DELIMITED="${RELEASE_ARCHIVES//,/ }"
+          for ARCHIVE in $SPACE_DELIMITED; do
+              echo "${ARCHIVE}"
+              ATTESTATION_FILE="$(basename "${ARCHIVE}").intoto.jsonl"
+              echo "Writing attestation to ${ATTESTATION_FILE}"
+              cat ${{ steps.attest_release_archive.outputs.bundle-path }} | jq --compact-output > "${ATTESTATIONS_DIR}/${ATTESTATION_FILE}"
+          done
+          echo "release_archive_attestations_dir=${ATTESTATIONS_DIR}" >> $GITHUB_OUTPUT
 
       - name: Release
         uses: softprops/action-gh-release@v1
```

Comment:
I think you can remove this dependency on softprops release gh action and just use gh cli directly.. kinda reduces that dependency and who has access to some of these write tokens.

Reply:
That's certainly possible. I'll just point out that this is a well-regarded action that's very popular, and I think it offers some nice functionality around overwriting release artifacts which is necessary if re-running the publish workflow.

Reply:
The community has also been using the `generates_release_notes` feature for a while to generate the change logs.

Reply:
kk sounds good